neon_core/gui/resting_screen.py

```diff
@@ -0,0 +1,232 @@
+        # Some skill other than the handler is showing a page
+        self.has_show_page = True
+
+        # If a skill overrides the animations do not show any
+        override_animations = message.data.get("__animations", False)
+        if override_animations:
+            # Disable animations
+            LOG.info("Disabling all animations for page")
+            self.override_animations = True
+        else:
+            LOG.info("Displaying all animations for page")
+            self.override_animations = False
+
+        # If a skill overrides the idle do not switch page
+        override_idle = message.data.get("__idle")
+        if override_idle is True:
+            # Disable idle screen
+            LOG.info("Cancelling Idle screen")
+            self.cancel_idle_event()
+            self.override(message)
+        elif isinstance(override_idle, int) and override_idle is not False:
+            LOG.info(
+                "Overriding idle timer to" " {} seconds".format(override_idle)
+            )
+            self.override(None)
+            self.start_idle_event(override_idle)
+        elif message.data["page"] and not message.data["page"][0].endswith(
+                "idle.qml"
+        ):
+            # Check if the show_page deactivates a previous idle override
+            # This is only possible if the page is from the same skill
+            LOG.info("Cancelling idle override")
+            if override_idle is False and compare_origin(
+                    message, self.override_idle[0]
+            ):
+                # Remove the idle override page if override is set to false
+                self.cancel_override()
+            # Set default idle screen timer
+            self.start_idle_event(30)
+
+    def cancel_idle_event(self):
+        """Cancel the event monitoring current system idle time."""
+        self.next = 0
+        self.cancel_scheduled_event("IdleCheck")
+
+    def start_idle_event(self, offset=60, weak=False):
+        """Start an event for showing the idle screen.
+        Arguments:
+            offset: How long until the idle screen should be shown
+            weak: set to true if the time should be able to be overridden
+        """
+        with self.lock:
+            if time.monotonic() + offset < self.next:
+                LOG.info("No update, before next time")
+                return
+
+            LOG.info("Starting idle event")
+            try:
+                if not weak:
+                    self.next = time.monotonic() + offset
+                # Clear any existing checker
+                self.cancel_scheduled_event("IdleCheck")
+                time.sleep(0.5)
+                self.schedule_event(
+                    self.show, int(offset), name="IdleCheck"
+                )
+                LOG.info("Showing idle screen in " "{} seconds".format(offset))
+            except Exception as e:
+                LOG.exception(repr(e))
+
+
+def init_resting_screen_manager():
```

Comment:
could this be integrated instead into the GUIManager class? i think that would feel cleaner

Reply:
Good catch; missed that cleanup after porting from the skill